src/io/flutter/perf/PerfService.java

```diff
@@ -35,6 +37,16 @@
 
   private boolean isRunning;
 
+  public static void updateRegisteredExtensions(Isolate isolate) {
+    List<FlutterApp> apps = FlutterApp.collectAllApps();
+    for (FlutterApp app : apps) {
+      PerfService service = app.getPerfService();
+      if (service != null) {
+        service.addRegisteredExtensionRPCs(isolate);
```

Comment:
this doesn't seem right. The isolate likely is not a valid isolate for most of these apps.

Reply:
Good point. I need to re-think this bit.